.github/workflows/zulip-ci.yml

```diff
@@ -35,6 +29,18 @@ jobs:
             os: jammy
             legacy_client_interface: "6"
             server_version: refs/tags/6.0
+          # Debian 12 ships with Python 3.11.2.
+          - docker_image: zulip/ci:bookworm
+            name: Debian 12 (Python 3.11, backend + documentation)
+            os: bookworm
+            legacy_client_interface: "7"
+            server_version: refs/tags/7.0
+          # Ubuntu 24.04 ships with Python 3.12.3.
+          - docker_image: zulip/ci:noble
+            name: Ubuntu 24.04 (Python 3.12, backend)
+            os: noble
+            legacy_client_interface: "8"
+            server_version: refs/tags/8.5
```

Comment:
Earliest minor Zulip version that [supports](https://github.com/zulip/zulip/blob/8.3/tools/lib/provision.py#L89) the platform is 8.3.

But, 8.3 and 8.4 are unable to use `pip` inside the venv, due to version conflicts related to `pkg_resources`. Only versions 8.5 onwards seem to be working.

<details>
<summary>Traceback</summary>

```
Traceback (most recent call last):
  File "/srv/zulip-py3-venv/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/cli/main.py", line 10, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/cli/autocompletion.py", line 9, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/cli/main_parser.py", line 7, in <module>
    from pip._internal.cli import cmdoptions
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/cli/cmdoptions.py", line 25, in <module>
    from pip._internal.cli.progress_bars import BAR_TYPES
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/cli/progress_bars.py", line 12, in <module>
    from pip._internal.utils.logging import get_indentation
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/utils/logging.py", line 18, in <module>
    from pip._internal.utils.misc import ensure_dir
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_internal/utils/misc.py", line 21, in <module>
    from pip._vendor import pkg_resources
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3254, in <module>
    @_call_aside
     ^^^^^^^^^^^
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3238, in _call_aside
    f(*args, **kwargs)
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3280, in _initialize_master_working_set
    tuple(
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3281, in <genexpr>
    dist.activate(replace=False)
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2788, in activate
    declare_namespace(pkg)
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2285, in declare_namespace
    _handle_ns(packageName, path_item)
  File "/srv/zulip-venv-cache/e2c2c053635e882ff393067adf908813bcad92ef/zulip-py3-venv/lib/python3.12/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2202, in _handle_ns
    loader = importer.find_module(packageName)
             ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'FileFinder' object has no attribute 'find_module'
```

</details>